src/_hmmc.cpp

```diff
@@ -26,7 +27,33 @@ double logsumexp(double const* v, ssize_t n)
   return std::log(acc) + max;
 }
 
-std::pair<py::array_t<double>, py::array_t<double>> forward_scaling(
+void log_mask_zero_1d(py::array_t<double> probs_,
```

Comment:
You could just return a `py::array_t<double>` here (allocating it from inside the function) instead of allocating it outside and modifying it in-place.

Reply:
So I had this originally, and then saw a strange issue with memory being reused when computing the `log_xi_sums` below.  However, not doing something like:
`auto log_transmat = log_mask_zero_2d(transmat).unchecked<2>()` and instead doing:

`auto log_transmat_ = log_mask_zero_2d(transmat);
auto log_transmat = log_transmat_.unchecked<2>()`
worked.